.github/workflows/build.yml

```diff
@@ -14,33 +14,52 @@ on:
       - '**.svg'
       - '**.drawio'
       - '.spelling'
+permissions:  # added using https://github.com/step-security/secure-workflows
+  contents: read
+
 jobs:
   skip-check:
+    permissions:
+      actions: write  # for fkirc/skip-duplicate-actions to skip or stop workflow runs
+      contents: read  # for fkirc/skip-duplicate-actions to read and compare commits
     runs-on: ubuntu-latest
     name: Skip the job?
     outputs:
           should_skip: ${{ steps.skip_check.outputs.should_skip }}
     steps:
+    - name: Harden Runner
+      uses: step-security/harden-runner@2e205a28d0e1da00c5f53b161f4067b052c61f34
+      with:
+        egress-policy: audit # TODO: change to 'egress-policy: block' after couple of runs
+
     - id: skip_check
-      uses: fkirc/skip-duplicate-actions@v5.2.0
+      uses: fkirc/skip-duplicate-actions@f11521568414503656a5af807dc3018c012552c4
```

Comment:
these are good changes, but it would be nice if we still know to what version it corresponds. Going to the repo and finding the sha is too many steps. I can check how hard it would be to open PR towards https://github.com/step-security/secure-workflows to do that.. I mean to put the human readable x.y.z version to a comment

Reply:
related stuff https://github.com/step-security/secure-workflows/issues/1087#issuecomment-1242410647 tl;dr: they had it there, but dependabot removes the comment so they choose to not add it at all